pkg/status/status.go

```diff
@@ -93,22 +93,46 @@ func SetValidationStatus(s *common.Status, validationState []*common.ValidationS
 }
 
 func GetValidationStatus(s *common.Status) []*common.ValidationState {
-	return s.GetValidationStatus().GetValidationState()
+	validationStatus := s.GetValidationStatus()
+	if validationStatus != nil {
+		return validationStatus.GetValidationState()
+	}
+	return nil
 }
 
 // GetStatuses - converts status to map
 func GetStatuses(status *common.Status) map[string]*string {
 	statuses := map[string]*string{}
-	pstate := status.GetProvisioningStatus().String()
-	statuses["ProvisionState"] = &pstate
-	hstate := status.GetHealth().String()
-	statuses["HealthState"] = &hstate
-	estate := status.GetLastError().String()
-	statuses["Error"] = &estate
-	version := status.GetVersion().Number
-	statuses["Version"] = &version
-	dstate := status.GetDownloadStatus().String()
-	statuses["DownloadStatus"] = &dstate
+
+	provisionStatus := status.GetProvisioningStatus()
+	if provisionStatus != nil {
```

Comment:
We are skipping (not adding into the map) if provisionStatus== nil? Is everyone aware of status map may not contains some statuses. or should we add the empty string for provisionStatus? Same approach should apply next 4 statuses.

Reply:
The statuses are initialized on line 105 and it will contain default values of nil if the provisioning state is not available. This code is just added as a failsafe (or null check) and we are not skipping any valid provisioning state.